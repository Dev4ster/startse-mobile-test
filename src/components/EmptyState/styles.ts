import styled, { css } from 'styled-components/native';

export const Container = styled.View`
  padding: 10px;
`;

export const Title = styled.Text`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: bold;
    color: ${theme.colors.primary};
  `};
`;

export const ButtonContainer = styled.View`
  margin: 10px 0px;
  height: 30px;
  justify-content: center;
  align-items: center;
`;

export const Message = styled.Text``;
