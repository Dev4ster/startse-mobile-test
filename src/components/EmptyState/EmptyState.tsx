import React from 'react';
import i18n from '~/i18n';
import Button from '../Button/Button';

import * as S from './styles';

type EmptyStateProps = {
  title: string;
  message?: string;
  retry?: () => void;
};
const EmptyState = ({ title, message, retry }: EmptyStateProps) => {
  return (
    <S.Container>
      <S.Title>{title}</S.Title>
      {!!message && <S.Message>{message}</S.Message>}
      {!!retry && (
        <S.ButtonContainer>
          <Button onPress={retry} icon="refresh">
            {i18n.t('default.emptyState.retry')}
          </Button>
        </S.ButtonContainer>
      )}
    </S.Container>
  );
};

export default EmptyState;
