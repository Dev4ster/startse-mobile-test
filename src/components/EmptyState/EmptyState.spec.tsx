import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import EmptyState from './EmptyState';
import { renderWithTheme } from '~/utils/test/helpers';

describe('<EmptyState />', () => {
  it('should be match snapshot', () => {
    const { toJSON } = renderWithTheme(
      <EmptyState title="Error na aplicação" />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be render correctly', () => {
    const { getByText } = renderWithTheme(
      <EmptyState title="Error na aplicação" />,
    );
    expect(getByText(/Error na aplicação/)).toBeTruthy();
  });

  it('should be render button to retry', () => {
    const { getByText } = renderWithTheme(
      <EmptyState title="Error na aplicação" retry={() => true} />,
    );

    expect(getByText(/Tentar novamente/)).toBeTruthy();
  });

  it('should be call retry onPress button', () => {
    const mockFn = jest.fn();

    const { getByTestId } = renderWithTheme(
      <EmptyState title="Error na aplicação" retry={mockFn} />,
    );

    fireEvent(getByTestId('button'), 'onPress', null);
    expect(mockFn).toBeCalledTimes(1);
  });

  it('should be render message if prop is passed', () => {
    const { getByText } = renderWithTheme(
      <EmptyState
        title="Error na aplicação"
        message="mensagem padrão de erro"
      />,
    );

    expect(getByText(/mensagem padrão de erro/)).toBeTruthy();
  });
});
