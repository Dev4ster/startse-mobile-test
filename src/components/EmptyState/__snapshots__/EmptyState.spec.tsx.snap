// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should be match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#0080ed",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Error na aplicação
  </Text>
</View>
`;
